### DISCUSSION TITLE
Entitäten mit Unbekannten Daten

### DISCUSSION THREAD

**Discussion by Ingo-Vetter** [nathanmarlor/foxess_modbus#613] [Support]
Hallo, 
ich habe jetzt zwei H3 integriert und es läuft soweit alles, bis auf folgende Entitäten:

sensor.pv_power_wr_hausdach_master  
sensor.wr_hausdach_master_pv1_energy_total
sensor.wr_hausdach_master_pv2_energy_total

sensor.pv_power_wr_garage_slave
sensor.wr_garage_slave_pv1_energy_total
sensor.wr_garage_slave_pv2_energy_total

Hier bekomme ich keine Werte (Unbekannt)

Alle anderen Entitäten geben daten zurück. 

Was kann ich hier machen, damit die Entitäten Daten zurückgeben?

Viele Grüße
Ingo

**Comment by FozzieUK** [comment]
There is a `sensor.xxxxxx_solar_energy_today` which is derived from the hybrid inverters register for all PV the inverter has generated today.

There should also be 'total' sensors for PV1 & PV2 called

`sensor.xxxxxx_pv1_energy_total`
and
`sensor.xxxxxx_pv2_energy_total`

These are calculated within the integration and report the total 'Riemann' sum of PV1 & PV2 power measured by the integration, if you want daily totals you will need to use these as sources for a utility meter in home assistant - as they are calculated 'sums' it can take a few hours before any data will show once the integration has been installed.

I think on the pv_power you have the name and entity reversed - shouldn't they be named ?

`sensor.wr_hausdach_master_pv_power`
and
`sensor.wr_garage_slave_pv_power`

In home assistant if you go to Settings, Devices & Services, click on the 'x Device' under FoxESS Modbus integration that should show you all of the sensors being provided by the integration - but some are disabled by default.

Look at the bottom of the FoxESS Modbus integrations page and it will say '+xx entities not shown', if you click on that you can then select and enable any missing entities you want to be included.

**Comment by Ingo-Vetter** [comment]
Hallo,  dass mit den Namen kam von selbst.
Beim Einrichten der Wechselrichter habe ich z.B. WR Hausdach Master eingegeben damit man weis, das es der WR vom Hausdach ist und das es der Master ist.

Das mit den berechneten Summen habe ich mir schon gedacht, aber da scheint was nicht zu funktionieren, denn die Sensoren, welche die Daten aus den WR liefern, enthalten ja Daten... 

Beim WR Hausdach Master wären das die Sensoren:
sensor.wr_hausdach_master_pv1_power 
und 
sensor.wr_hausdach_master_pv2_power

![image](https://github.com/nathanmarlor/foxess_modbus/assets/26342084/1f4b612b-069f-4905-b52d-485b07bdfc08)

In welchem Unterordner finde ich den Code hierfür?
Eventuell gibt es einen Konflikt mit der Bezeichnung...

**Reply by FozzieUK** [reply]
On your home assistant can you go to Settings, Devices & Services, click on Foxess Modbus and then the device for ‘ WR Hausdach Master’

It will list all of your entities like this -
![IMG_1591](https://github.com/nathanmarlor/foxess_modbus/assets/113460294/0601ba60-dd4a-4c29-8750-60989585b173)
![IMG_1592](https://github.com/nathanmarlor/foxess_modbus/assets/113460294/4dbafb3f-3807-48f8-9d63-697818dc3ef2)
![IMG_1593](https://github.com/nathanmarlor/foxess_modbus/assets/113460294/58ee6935-1dd5-4c7d-8644-a6bfa79dc004)
![IMG_1594](https://github.com/nathanmarlor/foxess_modbus/assets/113460294/26be3761-1cca-45d9-b5df-3d78cfa586d9)

I’m afraid that it’s going to be multiple screen shots, but I need to see a list of all the entities being provided by the integration with their state.

**Reply by Ingo-Vetter** [reply]
Ich habe den Fehler gefunden...

Ich habe vor ein paar Wochen mal versucht, [HA-FoxESS-Modbus](https://github.com/StealthChesnut/HA-FoxESS-Modbus)
Öffentlich manuell zu integrieren (da wusste ich noch nicht dass ich einen Converter benötige).
Ich habe zwar ein Backup (ohne das ganze Modbus.. ) wieder hergestellt, aber, dass war wohl doch nicht ganz sauber.

Ich habe die Datei jetzt gelöscht und HA neu gestartet.
Jetzt sind alle Werte vorhanden.

Danke für die Unterstützung

**Reply by FozzieUK** [reply]
Great, glad you’ve got it sorted 👍
